File: teledonJFX/src/main/java/com/example/teledonjfx/repository/ICrudRepository.java
package com.example.teledonjfx.repository;

public interface ICrudRepository<T,Tid>{
    void save(T elem);
    void delete(Tid id);
    void update(T elem,Tid id);
    T findById(Tid id);
    Iterable<T> findAll();
    int size();
}

File: SocketTeledon/persistence/ICrudRepository.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace teledonCS.repository
{
    public interface ICrudRepository<Key,Element>
    {
        void save(Element el);
        void delete(Key key);
        void update(Element el);
        Element findOne(Key key);
        List<Element> findAll();
    }
}
